src/Utils/Utils.php

```diff
@@ -158,24 +158,31 @@ public static function getCleanShopwareVersion(): string
      */
     protected static function getShopwareVersion(): string
     {
+        $version = null;
         // Composer 2 runtime API uses the `InstalledVersions::class` in favor of the
         // deprecated/removed `Versions::class`
         if (class_exists(InstalledVersions::class)) {
-            $platformIsInstalled = InstalledVersions::isInstalled('shopware/platform');
-            if ($platformIsInstalled) {
-                return InstalledVersions::getPrettyVersion('shopware/platform');
+            if (InstalledVersions::isInstalled('shopware/platform')) {
+                $version = $version ?: InstalledVersions::getPrettyVersion('shopware/platform');
             }
 
-            return InstalledVersions::getPrettyVersion('shopware/core');
+            if (InstalledVersions::isInstalled('shopware/core')) {
+                $version = $version ?: InstalledVersions::getPrettyVersion('shopware/core');
+            }
         }
 
-        $packageVersions = Versions::VERSIONS;
-        $platformIsInstalled = isset($packageVersions['shopware/platform']);
-        if ($platformIsInstalled) {
-            return $packageVersions['shopware/platform'];
+        if (defined('PackageVersions\Versions::VERSIONS')) {
+            $packageVersions = Versions::VERSIONS;
+            if (isset($packageVersions['shopware/platform'])) {
+                $version = $version ?: $packageVersions['shopware/platform'];
```

Comment:
Overriding an already set variable, is screaming for hard debugging time. Please always prefer an early-return or declare a different variable, but don't change the value of an variable that already has a value set before. 

Here is a great article I have found about reassigning variable values https://zellwk.com/blog/dont-reassign/. It talks about JavaScript, but the same principle applies to PHP and many other languages aswell.